### ISSUE TITLE
New calendar is created daily basis

### ISSUE THREAD

**Issue by snsmakel** [derekantrican/GAS-ICS-Sync#242]
- Description of the problem

I've added my work calendar with ICS -link. Calendar is published in outlook.office365.com and given enough public rights. I created this calendar at first manually with same name than I added to Code.gs. Now it appears that my calendar gets new incarnation every day with same name and new color.

Calendar which I created manually is on part "Other calendars", but this GAS-ICS-Sync created these new calendars to "My Calendars" part. I have not tested whether GAS-ICS-Sync syncs all these calendars yet.

- Screenshots

I think at now there is no screenshot need.

- An input ics string that can reproduce the problem

I will deliver this ics link later (not publicly :)).

**Comment by derekantrican** [comment]
Rather than sharing the link, you can use derekantrican.com/icssanitizer to remove sensitive data and share with us

**Comment by snsmakel** [comment]
Hi,

I've put my calendar to sanitizer. How long it is supposed to run?

Here is snippet of code (scrambled link to broken)

Regards
 Severi
Code.gs rows 19 -> 49
*=========================================
*               SETTINGS
*=========================================
*/

var sourceCalendars = [                // The ics/ical urls that you want to get events from along with their target calendars (list a new row for each mapping of ICS url to Google Calendar)
                                       // For instance: ["https://www.calendarlabs.com/ical-calendar/ics/76/US_Holidays.ics", "US Holidays"]
  ["https://outlook.office365.com/owa/calendar/ZZZZZ@domain.fi/XXXXXXXXXX/calendar.ics", "MyName MyOrganization"],
  ["icsUrl2", "targetCalendar2"],
  ["icsUrl3", "targetCalendar1"]
  
];

var howFrequent = 15;                  // What interval (minutes) to run this script on to check for new events
var onlyFutureEvents = false;          // If you turn this to "true", past events will not be synced (this will also removed past events from the target calendar if removeEventsFromCalendar is true)
var addEventsToCalendar = true;        // If you turn this to "false", you can check the log (View > Logs) to make sure your events are being read correctly before turning this on
var modifyExistingEvents = true;       // If you turn this to "false", any event in the feed that was modified after being added to the calendar will not update
var removeEventsFromCalendar = true;   // If you turn this to "true", any event created by the script that is not found in the feed will be removed.
var addAlerts = true;                  // Whether to add the ics/ical alerts as notifications on the Google Calendar events, this will override the standard reminders specified by the target calendar.
var addOrganizerToTitle = false;       // Whether to prefix the event name with the event organiser for further clarity
var descriptionAsTitles = false;       // Whether to use the ics/ical descriptions as titles (true) or to use the normal titles as titles (false)
var addCalToTitle = false;             // Whether to add the source calendar to title
var addAttendees = false;              // Whether to add the attendee list. If true, duplicate events will be automatically added to the attendees' calendar.
var defaultAllDayReminder = -1;        // Default reminder for all day events in minutes before the day of the event (-1 = no reminder, the value has to be between 0 and 40320)
                                       // See https://github.com/derekantrican/GAS-ICS-Sync/issues/75 for why this is neccessary.
var overrideVisibility = "";           // Changes the visibility of the event ("default", "public", "private", "confidential"). Anything else will revert to the class value of the ICAL event.
var addTasks = false;

var emailSummary = false;              // Will email you when an event is added/modified/removed to your calendar
var email = "";                        // OPTIONAL: If "emailSummary" is set to true or you want to receive update notifications, you will need to provide your email address

**Comment by snsmakel** [comment]
Hi, 

Sanitizer is still running.. Almos an hour already. My calendar is quite big, should I stop it?

Regards
 Severi

**Comment by snsmakel** [comment]
Hi, 

Sanitizer is still going on status "Working..." . It has been there for several hours now. I think it will not get ready. My work calendar has quite much recurring events and also one time events.

I stopped sanitizer now.

Regards
 Severi

**Comment by jonas0b1011001** [comment]
> Calendar which I created manually is on part "Other calendars"

Calendars listed as "other calendars" are _read only_, the script can not edit these events which is why a new calendar with read/write permissions is created.

How often does it create duplicates? on each run or just occasionally?

A few days ago i also realized that a duplicate calendar was created, that happened only once though. It was quite certain caused by an incomplete response by the API.

**Comment by snsmakel** [comment]
> 

Hi,

Yes I understand this "other calendar" part. So far I've noted this duplication once. 

I've used script now only 4 days. Script has created two calendars, one just after first run and this second one one or two days after that. I've had so much happenings in work and private life at this week that days got confused in my mind :).

This "doubling" is quite confusing because we have common calendar and backup account on gmail for me and my wife. Wife suddenly noted that she has calendar full of my work appointments.. This has happened now twice, but I think first is the time when script started to work and second appeared one or two days ago.

Regards
 Severi

**Comment by derekantrican** [comment]
Is this creating a _new calendar_? Or _duplicating events on the same calendar_? It also sounds like you have the same calendar in "other calendars" (synced by Google) and in "my calendars" (synced by the script). I would suggest removing or hiding (unchecking) the one under "other calendars" as that might cause a bit of visual confusion too.

A couple side notes:

- If the sanitizer doesn't complete within a couple minutes, feel free to close it. Not sure what would cause it to hang/error like that but you definitely shouldn't need to wait an hour for it.
- You can remove the last two calendar entries from the script (the lines that say "icsUrl2" and "icsUrl3"). Those are just examples. Just remember to remove the comma at the end of the line for your calendar group

**Comment by snsmakel** [comment]
Hi,

It was creating a new instance of this same calendar, with same name in "my calendar" part. There was then 2 of "My Work" -calendars at the same time at "my calendar" and one in "other calendars". I removed these extras then and so far there has not been created a new "My Work" -calendar yet! :)

I made short review of old cases here and thin it might be something to do with too often triggered update of calendar or manual runs. Cases were discussed in Discord at May 2021. There was 1 minute update frequency. I might have run the script manually when it might be running background at the same time. 

I've removed now these icsUrl2 and 3 from Code.gs. Let's see what happens in near future.

Regards
 Severi
